check_all_options.c

```diff
@@ -101,6 +101,30 @@ void check_options_tcp_rr(struct options *opts, struct callbacks *cb)
 
 void check_options_tcp_stream(struct options *opts, struct callbacks *cb)
 {
+#ifdef WITH_TCPDIRECT
+      if (opts->tcpd_gpu_pci_addr) {
+            CHECK(cb, opts->tcpd_nic_pci_addr,
+                  "Must provide NIC PCI address if GPU PCI address was provided.");
+            CHECK(cb, opts->tcpdirect_phys_len > 0,
+                  "Must provide non-zero --tcpdirect-phys-len flag if GPU PCI address was provided.");
+            // TODO check page-alignment
```

Comment:
Please remove all commented out, dead code, etc.

To put things into perspective, this is a high priroity, visible effort, and the cod equality should be representive. Soon, I'm going to start pointing upstream kernel reviewers to this neper support, and we're going to ship it to Canonical (Ubuntu providers, etc). The functionality is great but lets make sure the code quality is also utmost!